smart-contracts/contracts/cl-vault/src/vault/range.rs

```diff
@@ -418,22 +479,21 @@ pub fn do_swap_deposit_merge(
     let swap_params = SwapParams {
         token_in_amount: swap_amount,
         token_out_min_amount,
-        token_in_denom,
+        token_in_denom: token_in_denom.clone(),
         token_out_denom,
         recommended_swap_route: mrs.recommended_swap_route,
         force_swap_route: mrs.force_swap_route,
     };
 
-    let token_in_denom = swap_params.token_in_denom.clone();
-
-    let swap_msg = swap(deps, &env, swap_params)?;
+    let swap_msg = swap_msg(deps, &env, swap_params)?;
 
-    Ok(Response::new()
-        .add_submessage(SubMsg::reply_on_success(swap_msg, Replies::Swap.into()))
-        .add_attribute("method", "reply")
-        .add_attribute("action", "do_swap_deposit_merge")
-        .add_attribute("token_in", format!("{}{}", swap_amount, token_in_denom))
-        .add_attribute("token_out_min", format!("{}", token_out_min_amount)))
+    Ok(SwapCalculationResult {
```

Comment:
This struct seems overly complex to me with the `Option` for `swap_msg` and `token_in_denom`. I'd expect this to be replaceable by just returning an Option`SwapCalculationResult` and using the option to signifiy whether a swap should happen